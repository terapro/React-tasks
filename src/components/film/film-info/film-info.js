import React from 'react';
import PropTypes from 'prop-types';

import {Caption} from 'src/components/common/caption/caption.js';
import {YearMinutes} from "src/components/film/year-minutes/year-minutes";
import {FilmHeader} from "src/components/film/film-header/film-header.js";

const FilmInfo = ({info}) => {
  return (
    <div className='film-info'>
      <FilmHeader title={info['title']} rating={+info['vote_average']}/>
      <Caption content={info['tagline']} txtWhite txtItalic />
      <YearMinutes minutes={info['runtime']} year={info['release_date']}/>
      <Caption content={info['overview']} txtWhite />
    </div>
  );
};

FilmInfo.propTypes ={
  info: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    tagline: PropTypes.string,
    vote_average: PropTypes.number,
    vote_count: PropTypes.number,
    release_date: PropTypes.string,
    poster_path: PropTypes.string,
    overview: PropTypes.string,
    budget: PropTypes.number,
    revenue: PropTypes.number,
    genres: PropTypes.arrayOf(PropTypes.string),
    runtime: PropTypes.number
  }).isRequired
};

export {FilmInfo};