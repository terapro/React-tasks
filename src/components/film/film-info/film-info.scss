.film-info {
  width: $filmInfoWidth;
  padding: 0 $elementPadding 0 0;

}