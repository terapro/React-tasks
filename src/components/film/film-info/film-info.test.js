import React from 'react';
import {FilmInfo} from 'src/components/film/film-info/film-info.js';
import {shallow} from 'enzyme';

const mockFilm = {
  "id": 333339,
  "title": "Ready Player One",
  "tagline": "A better reality awaits.",
  "vote_average": 8.1,
  "vote_count": 617,
  "release_date": "2018-03-28",
  "poster_path": "https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg",
  "overview": "When the creator of a popular video game system dies, a virtual contest is created to compete for his fortune.",
  "budget": 175000000,
  "revenue": 0,
  "genres": [
    "Adventure",
    "Science Fiction",
    "Action"
  ],
  "runtime": 140
};

describe('<FilmInfo />', () => {
  describe('Rendering', () => {
    it('should render the component', () => {
      const wrapper = shallow(<FilmInfo info={mockFilm}/>);
      expect(wrapper).toMatchSnapshot();
    });
  });
});