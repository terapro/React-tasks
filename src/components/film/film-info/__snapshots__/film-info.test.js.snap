// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmInfo /> Rendering should render the component 1`] = `
<div
  className="film-info"
>
  <FilmHeader
    rating={8.1}
    title="Ready Player One"
  />
  <Caption
    content="A better reality awaits."
    txtItalic={true}
    txtWhite={true}
  />
  <YearMinutes
    minutes={140}
    year="2018-03-28"
  />
  <Caption
    content="When the creator of a popular video game system dies, a virtual contest is created to compete for his fortune."
    txtItalic={false}
    txtWhite={true}
  />
</div>
`;
